llvm/include/llvm/MC/MCRegisterInfo.h

```diff
@@ -726,60 +730,122 @@ class MCRegUnitRootIterator {
 /// MCRegAliasIterator enumerates all registers aliasing Reg.  If IncludeSelf is
 /// set, Reg itself is included in the list.  This iterator does not guarantee
 /// any ordering or that entries are unique.
+///
+/// This iterator can work in two modes: cached and uncached.
```

Comment:
The fact that cached and uncached implementations are interleaved in one class seems pretty awkward. Might it make sense to only expose the cached version here and have the uncached version inside getCachedAliasesOf() only?

Reply:
I did that first because I assumed we'd like a killswitch of some kind for targets where the cache doesn't help, but if I read the compile time tracker diff correctly, it seems to be a universal win so I'll revisit it and split up the implementation
![image](https://github.com/llvm/llvm-project/assets/29600849/48ea789b-4e4c-4a74-8df6-b800541275e2)